### PR TITLE
nasm: Conan 2 compatibility tweaks

### PR DESCRIPTION
**Pull request by System-Arch**
Specify library name and version:  **nasm/2.15.05**

Minor tweaks to enable test to pass with Conan 2 and take advantage of latest bug fixes.

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by System-Arch** [comment]
> * [x]  I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.

"Latest" is currently 1.54.0 but CCI appears to still be using 1.53.0. Perhaps automation is needed to pull in the latest Conan release.

File: recipes/nasm/all/conanfile.py [hunk 1]
```diff
@@ -49,12 +51,8 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def generate(self):
-        env = VirtualBuildEnv(self)
-        env.generate()
```
Comment (SpaceIm) [inline recipes/nasm/all/conanfile.py:53]: must not be removed

File: recipes/nasm/all/conanfile.py [hunk 2]
```diff
@@ -92,13 +89,12 @@ def package(self):
...
-            autotools.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])
+            autotools.install()
             rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         self.cpp_info.libdirs = []
         self.cpp_info.includedirs = []
 
         # TODO: Legacy, to be removed on Conan 2.0
-        self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
```
Comment (SpaceIm) [inline recipes/nasm/all/conanfile.py:104]: must not be removed

File: recipes/nasm/all/conanfile.py [hunk 3]
```diff
@@ -92,13 +89,12 @@ def package(self):
...
+        self.runenv_info.append_path("PATH", os.path.join(self.package_folder, "bin"))
```
Comment (SpaceIm) [inline recipes/nasm/all/conanfile.py:100]: no, bindirs is automatically added to PATH of VirtualBuildEnv

File: recipes/nasm/all/conanfile.py [hunk 4]
```diff
@@ -64,12 +62,11 @@ def generate(self):
 
     def build(self):
         apply_conandata_patches(self)
-
+        autotools = Autotools(self)
         if is_msvc(self):
             with chdir(self, self.source_folder):
-                self.run("nmake /f {}".format(os.path.join("Mkfiles", "msvc.mak")))
+                autotools.make(f'/f {os.path.join("Mkfiles", "msvc.mak")}')
```
Comment (SpaceIm) [inline recipes/nasm/all/conanfile.py:68]: doesn't make sense

Comment (System-Arch) [reply recipes/nasm/all/conanfile.py:68]: Hi @SpaceIm,

Thanks for the review comments. The code for Autotools makes explicit mention of nmake so I have been assuming that it is designed to handle it: See Conan2.0 source file conan/tools/gnu/autotools.py:

>         if "-j" not in str_args and "nmake" not in make_program.lower():

File: recipes/nasm/all/conanfile.py (hunk 2 above)
Comment (System-Arch) [reply recipes/nasm/all/conanfile.py:104]: Hi @SpaceIm,

I thought:

> self.runenv_info.append_path("PATH", os.path.join(self.package_folder, "bin"))

was the Conan 2 compatible way fo saying the same thing. No?

File: recipes/nasm/all/conanfile.py (hunk 1 above)
Comment (System-Arch) [reply recipes/nasm/all/conanfile.py:53]: Hi @SpaceIm,

What purpose does this serve when doing automated builds? They seem to work fine without it and I was just trying to eliminate extraneous code. Thanks

File: recipes/nasm/all/conanfile.py (hunk 2 above)
Comment (System-Arch) [reply recipes/nasm/all/conanfile.py]: Restored now that Conan 2.0 doesn't error out on this construct.

File: recipes/nasm/all/conanfile.py (hunk 1 above)
Comment (System-Arch) [reply recipes/nasm/all/conanfile.py]: Restored as it seems to cause no harm but still unclear on why it is needed when things appear to work fine without it.

File: recipes/nasm/all/conanfile.py (hunk 4 above)
Comment (System-Arch) [reply recipes/nasm/all/conanfile.py:71]: Hi @SpaceIm,
Okay, restored use of self.run as @prince-chrismc did not like using tools.gnu:make_program to specify nmake from within a recipe.

File: recipes/nasm/all/conanfile.py [hunk 5]
```diff
@@ -64,12 +63,11 @@ def generate(self):
 
     def build(self):
         apply_conandata_patches(self)
-
+        autotools = Autotools(self)
```
Comment (SpaceIm) [inline recipes/nasm/all/conanfile.py:66]: as I said it makes little sense, just wait NMakeToolchain of conan 1.55.0

**Comment by System-Arch** [comment]
Hi @prince-chrismc,
Congrats on getting the Conan 2.0 pipeline up and running. FWIW, this change appears to have failed in the Conan 2.0 pipeline because it is running with an outdated version of Conan 2.0 that lacks fix for conan-io/conan#12673. Any word on when the pipeline will be updated to the latest 2.0 release? Thanks

P.S. Also, the 2.0 failure says "Not required for merge"  Wouldn't it a good idea to start requiring recipes to pass with the latest version of 2.0 in order to be merged. That has certainly been my criteria before opening any PRs. Thanks.

**Comment by uilianries** [comment]
> Any word on when the pipeline will be updated to the latest 2.0 release? Thanks

@System-Arch We need to update Docker images internally, not same from ConanDockerTools, then, create new Jenkins labels, and update some ansible scripts. It may take one week or two.

**Comment by uilianries** [comment]
> P.S. Also, the 2.0 failure says "Not required for merge" Wouldn't it a good idea to start requiring recipes to pass with the latest version of 2.0 in order to be merged

@System-Arch No because most of dependencies are not available in ConanCenter prepared for Conan v2, so in case your PR depends on a package which is ported and uploaded, it will fail. Also, Conan v2 is still in Beta, so some features can be changed and it would not be great needing to change recipes in CCI because a new change. Anyway, in few months it will be mandatory. For more information, we have a roadmap documented here: https://github.com/conan-io/conan-center-index/blob/master/docs/v2_roadmap.md#long-term

File: recipes/nasm/all/conanfile.py (hunk 5 above)
Comment (System-Arch) [reply recipes/nasm/all/conanfile.py]: Hi @SpaceIm,
Okay, I've updated the recipe to use NMakeToolchain. Hopefully CCI will upgrade to 1.55.0 soon

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**